test/systemtests/k8setup_test.go

```diff
@@ -229,32 +257,76 @@ func (k *kubernetes) getMACAddr(c *container, dev string) (string, error) {
 	return out, err
 }
 
-func (k *kubernetes) exec(c *container, args string) (string, error) {
-	cmd := fmt.Sprintf("kubectl exec %s -- %s", c.containerID, args)
-	logrus.Infof("Exec: Running command %s", cmd)
+/*
+* exec is used to run a specific command using kubectl on the host
+ */
+func (k *kubernetes) exec(podName, args string, ns ...string) (string, error) {
```

Comment:
Can we change it back to container structure here ? For debuggability purpose , to keep it consistent with exec in other implementations and if we need to use specific pod related stuff like node info etc it is much better to pass the struct than the ID and this may not be buying anything for us.